get-extracts/get-extracts.py

```diff
@@ -1,146 +1,81 @@
-#organization's API key.
-
-username = 'MyApiKey'
-password = 'MyApiSecret'
-
-
-
-#By default, the script stores the extract files in the directory where the script is running
-#To specify a storage directory, change this value to a string represting the directory where
-#the files should be stored.
-#Example: storageDir = '/data/ookla/extracts'
+# Please replace MyApiKey and MyApiSecret below with your organization's API key.
+username = 'my_api_key'
+password = 'my_api_secret'
 
+# By default, the script stores the extract files in the directory where the script is running
+# To specify a storage directory, change this value to a string represting the directory where
+# the files should be stored.
+# Example: storageDir = '/data/ookla/extracts'
 storageDir = os.getcwd()
 
-opener = urllib2.build_opener()
-urllib2.install_opener(opener)
-
-#request json list of files
+opener = compatible_urllib.build_opener()
+compatible_urllib.install_opener(opener)
 opener.addheaders = [('Accept', 'application/json')]
 
-#setup authentication
-base64string = base64.b64encode('%s:%s' % (username, password))
+# setup authentication
+login_credentials = '%s:%s' % (username, password)
+base64string = base64.b64encode(login_credentials.encode('utf-8')).decode('ascii')
 opener.addheaders = [('Authorization', 'Basic %s' % base64string)]
 
-#If login page is returned, raise error message.
-result = urllib2.urlopen(theurl)
-if result.info().type == "text/html":
-    print "Authentication error.\nPlease verify that the API key is correct."
-    sys.exit()
-
+# makes request for files
+response = compatible_urllib.urlopen(extracts_url).read()
 try:
-    response = result.read()
-except urllib2.HTTPError, err:
-   if err.code == 500:
-       print "Error: The account associated with this API key does not have access to data extracts.\nPlease contact your technical account manager to enable data extracts for this account."
-       sys.exit()
-try:
-    json_data = json.loads(response)
-except ValueError, e:
-    print e
+    content = json.loads(response)
+except ValueError as err:
+    print(err)
     sys.exit()
 
+#############################################################
+# loop through contents, sort through files and directories
+def sort_files_and_directories(contents):
+    for entry in contents:
+        if entry['type'] == 'file' and entry['name'].find('headers') == -1 and '_20' in entry['name']:
+            filter(entry)
+        elif entry['type'] == 'dir':
+            subdir = extracts_url + entry['url']
+            files = json.loads(compatible_urllib.urlopen(subdir).read())
+            sort_files_and_directories(files)
+
+# determine if file should be downloaded - check for new datasets and most current file for exisiting datasets
+def filter(file):
+    # identify the dataset by the file name prefix
+    dataset = file['name'][:file['name'].index('_20')]
+    if dataset not in files or dataset in files and file['mtime'] > files[dataset]['age']:
```

Comment:
```python
dataset not in files or (dataset in files and file['mtime'] > files[dataset]['age'])
```
is equivalent to
```python
dataset not in files or file['mtime'] > files[dataset]['age']
```
right?
(if `dataset in files` were not true, it wouldn't get there anyways)